### PR TITLE
Framework Docs

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

Bindings and calls are described in detail. Install, provisioning, and authentication need to be fleshed out still.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-33252

### CODE REVIEW

File: docs/framework-docs.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+- RootPost - Root post of the selected post
+
+- Team - The current ream the acting user is interacting with
+
+
+### Conext Expansion Example
+
+In this example, the bindings specify to expand the post the user clicks on:
+
+![binding-form-diagram.png](https://user-images.githubusercontent.com/6913320/109165112-2e6ac800-7749-11eb-8d83-d495258f3f1e.png)
```
Comment (mickmister) [inline docs/framework-docs.md:107]: @larkox You said you were working on sequence diagrams. Is this something similar to what you are working on?

Comment (larkox) [reply docs/framework-docs.md:107]: Exactly that. Here are some that I did, but they are mainly "brainstorming".
https://docs.google.com/document/d/18fhHqLboWWQiMzY_LJ91PlK6Tw_ziapzpUmABajPSxY/edit?usp=sharing

File: docs/framework-docs.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
+## Provisioning
+- Packaging (http server -> Lambda function)
+- Static assets: icons only
```
Comment (hanzei) [inline docs/framework-docs.md:3]: Right now the app can declare any files as assets. Why should we limit that?

File: docs/framework-docs.md [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+  - Serving them out of S3
```
Comment (hanzei) [inline docs/framework-docs.md:4]: For HTTP Apps the App has to serve it themself via HTTP.

File: docs/framework-docs.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+  - Validate the contents to be harmless icons??
+
```
Comment (hanzei) [inline docs/framework-docs.md:6]: The URL of the assets is `$SITE_URL/plugins/com.mattermost.apps/api/v1/static/$APP_ID$/$ASSET_NAME`

File: docs/framework-docs.md [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+## Install
+- Manifest -> Installed App
+  - Consent to permissions, locations, OAuth app type
+  - Create Bot+Access Token, OAuth App
+  - HTTP: collect app’s JWT secret
+- Invoke “OnInstall” callback on the App
+  - Admin access token
+- Also Uninstall/Enable/Disable per App
```
Comment (hanzei) [inline docs/framework-docs.md:14]: Right now there is only uninstall. Should we make that clear?

File: docs/framework-docs.md [hunk 6]
```diff
@@ -0,0 +1,327 @@
+## Provisioning
+- Packaging (http server -> Lambda function)
+- Static assets: icons only
+  - Serving them out of S3
+  - Validate the contents to be harmless icons??
+
+## Install
+- Manifest -> Installed App
```
Comment (hanzei) [inline docs/framework-docs.md:8]: For HTTP Apps that is true. For AWS Apps we fetch the manifest from S3 and use that.

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for writing this overview :+1: 

Could you please add an introduction to the docs to help me and other understand who the audience is and what this document cover?

**Comment by mickmister** [comment]
> Could you please add an introduction to the docs to help me and other understand who the audience is and what this document cover?

@hanzei Will do. For a brief answer here, this doc is meant to help the security team review the framework. The idea was for me to submit a WIP that includes bindings and calls, and @levb (and others possibly?) to include additions for installing, provisioning, and authentication in this same PR.

The ultimate audience is probably App developers. I'll write an introduction as such, and continue documenting the interactivity pieces.

**Comment by mickmister** [comment]
Closing in favor of https://github.com/mattermost/mattermost-plugin-apps/pull/100

**Pull request closed without merging**